packages/flutter/lib/src/foundation/diagnostics.dart

```diff
@@ -514,11 +692,98 @@ final TextTreeConfiguration singleLineTextConfiguration = TextTreeConfiguration(
   lineBreakProperties: false,
   addBlankLineIfNoChildren: false,
   showChildren: false,
+  propertyPrefixIfChildren: '  ',
+  propertyPrefixNoChildren: '  ',
+  linkCharacter: '',
+  prefixOtherLinesRootNode: '',
+);
+
+/// Render a node as a single line omitting children styling the node like it is
+/// a header describing content following it in the tree even though the node is
+/// not actually the parent of the content following it.
+///
+/// Example:
+/// `<name>: <description>(<property1>, <property2>, ..., <propertyN>):`
+///
+/// See also:
+///
+///  * [DiagnosticsTreeStyle.headerLine]
+final TextTreeConfiguration headerLineTextConfiguration = TextTreeConfiguration(
+  propertySeparator: ', ',
+  beforeProperties: '(',
+  afterProperties: ')',
+  mandatoryAfterProperties: ':',  // Colon that makes this style look like a header.
+  prefixLineOne: '',
+  prefixOtherLines: '  ',
+  prefixLastChildLineOne: '',
+  lineBreak: '',
+  lineBreakProperties: false,
+  addBlankLineIfNoChildren: false,
+  showChildren: false,
   propertyPrefixIfChildren: '',
   propertyPrefixNoChildren: '',
   linkCharacter: '',
   prefixOtherLinesRootNode: '',
 );
+/// Render a node as a single line omitting children.
```

Comment:
The example below has two lines?

Reply:
Done. Updated the description and renamed to
errorProperty

Reply:
Updated the description to match the example.